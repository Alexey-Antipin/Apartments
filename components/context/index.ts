import { createContext } from "react";

type ContextGlobal = {
  setHeart: (value: any) => void;
  heart: number;
};

export const Context = createContext<ContextGlobal>({
  heart: 0,
  setHeart: () => {},
});
