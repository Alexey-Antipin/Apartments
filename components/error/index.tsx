import styles from "./Error.module.scss";
import { Sprite } from "../../svg";
import Image from "next/image";
import Link from "next/link";

export const Error: React.FC = () => {
  return (
    <div className={styles.error}>
      <div className={styles.wrapper}>
        {/* Точки жёлтые */}
        <Image
          className={styles.picture}
          src="/points.png"
          alt="points"
          height="60"
          width="60"
        />

        {/* Блок */}
        <div className={styles.block}>
          <h2 className={styles.title}>Ошибка 404</h2>
          <p className={styles.text}>
            Возможно, у вас опечатка в адресе страницы, или её просто не
            существует
          </p>
          <Link className={styles.button} href="/">
            <Sprite id="home" colour="#1E2123" />
            <span className={styles["button-text"]}>Вернуться на главную</span>
          </Link>
        </div>

        {/* Ошибка svg */}
        <div className={styles.sprite}>
          <Sprite id="error" />
        </div>

        {/* Точки белые */}
        <Image
          className={styles["picture-white"]}
          src="/white-points.png"
          alt="points"
          height="60"
          width="60"
        />
      </div>
    </div>
  );
};
