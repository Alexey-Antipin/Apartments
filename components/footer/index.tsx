import { FooterOfArrayList, Network } from "../../ts";
import styles from "./Footer.module.scss";
import { List } from "../../common";
import { Sprite } from "../../svg";
import Image from "next/image";
import Link from "next/link";

export const Footer: React.FC = () => {
  const socialNetwork: Network[] = [
    { name: "instagram", href: "https://www.instagram.com/" },
    { name: "vk", href: "https://vk.com/" },
    { name: "facebook", href: "https://ru-ru.facebook.com/" },
  ];
  const arraysList: FooterOfArrayList = {
    a: ["Коттеджи и усадьбы", "Бани и сауны", "Авто на прокат"],
    b: [
      "Квартиры",
      "Квартиры в Минске",
      "Квартиры в Гомеле",
      "Квартиры в Бресте",
    ],
    c: [
      "",
      "",
      "Квартиры в Витебске",
      "Квартиры в Гродно",
      "Квартиры в Могилеве",
    ],
    d: ["Новости", "Размещение и тарифы", "Объявления на карте", "Контакты"],
  };

  return (
    <footer className={styles.footer}>
      <div className={styles.info}>
        <div>
          <Link href="./">
            <Image src="/logo.png" height="20" width="134" alt="logo" />
          </Link>
          <h4 className={styles["title-address"]}>СДАЁМ БАЙ</h4>
        </div>

        <address className={styles.address}>
          ИП Шушкевич Андрей Викторович
          <br /> УНП 192602485 Минским горисполкомом
          <br />
          <time>10.02.2016</time>
          <br /> 220068, РБ, г. Минск, ул. Осипенко, 21, кв.23
          <br />
          <Link className={styles.telefon} href="tel:+375296214833">
            +375 29 621 48 33,
          </Link>
          <Link className={styles.colour} href="mailto:sdaem@sdaem.by">
            sdaem@sdaem.by
          </Link>
          <br /> Режим работы: <time>08:00-22:00</time>
        </address>
      </div>

      <div className={styles.container}>
        <nav className={styles.navigation}>
          <List
            classes={{
              classUl: styles["list-black"],
            }}
            array={arraysList.a}
          />
          <List
            classes={{
              classList: styles["item-second-block"],
            }}
            array={arraysList.b}
          />
          <List array={arraysList.c} />
          <List array={arraysList.d} />
        </nav>

        <div className={styles.block}>
          <ul className={styles["social-network"]}>
            <h3 className={styles.title}>Мы в соцсетях</h3>
            {socialNetwork.map((elem, index) => (
              <li key={index}>
                <Link className={styles.sprite} href={elem.href}>
                  <Sprite id={elem.name} />
                </Link>
              </li>
            ))}
          </ul>

          <ul className={styles.card}>
            {[
              { h: 13, w: 42 },
              { h: 13, w: 86 },
              { h: 16, w: 38 },
              { h: 20, w: 26 },
              { h: 16, w: 46 },
              { h: 30, w: 22 },
            ].map((elem, index) => (
              <li key={index}>
                <Image
                  src={`/card/i${index + 1}.png`}
                  height={elem.h}
                  width={elem.w}
                  alt="card"
                />
              </li>
            ))}
          </ul>
        </div>
      </div>
    </footer>
  );
};
