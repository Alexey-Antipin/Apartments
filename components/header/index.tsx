import { useEffect, useState } from "react";
import styles from "./Header.module.scss";
import { getCookie } from "cookies-next";
import { MassiveOfList, MassiveOfSelect } from "../../ts";
import { Sprite } from "../../svg";
import Image from "next/image";
import Link from "next/link";
import clsx from "clsx";
import { Select } from "../../common/Select";

export const Header: React.FC = () => {
  const [activeId, setActiveId] = useState<number>(0);
  const [cookie, setCookie] = useState<string | boolean>("");

  const link: MassiveOfList[] = [
    { id: 1, text: "Главная", href: "./" },
    { id: 2, text: "Новости", href: "./news" },
    { id: 3, text: "Размещение и тарифы", href: "./" },
    {
      id: 4,
      text: "Объявления на карте",
      href: "./",
      sprite: "sign",
    },
    { id: 5, text: "Контакты", href: "./contacts" },
  ];
  const bookmark = {
    id: 6,
    text: "Закладки",
    href: "./",
  };
  const underList: MassiveOfSelect[] = [
    {
      id: 7,
      text: "Квартиры на сутки",
      sprite: "sign",
      list: [
        { id: 1, text: "Квартиры на сутки в Минске" },
        { id: 2, text: "Квартиры на сутки в Гомеле" },
        { id: 3, text: "Квартиры на сутки в Бресте" },
        { id: 4, text: "Квартиры на сутки в Витебске" },
        { id: 5, text: "Квартиры на сутки в Гродно" },
        { id: 6, text: "Квартиры на сутки в Могилеве" },
      ],
    },
    {
      id: 8,
      text: "Коттеджи и усадьбы",
      list: [
        { id: 1, text: "Коттеджи на сутки в Минске" },
        { id: 2, text: "Коттеджи на сутки в Гомеле" },
        { id: 3, text: "Коттеджи на сутки в Бресте" },
        { id: 4, text: "Коттеджи на сутки в Витебске" },
        { id: 5, text: "Коттеджи на сутки в Гродно" },
        { id: 6, text: "Коттеджи на сутки в Могилеве" },
      ],
    },
    {
      id: 9,
      text: "Бани и Сауны",
      list: [
        { id: 1, text: "Бани и Сауны в Минске" },
        { id: 2, text: "Бани и Сауны в Гомеле" },
        { id: 3, text: "Бани и Сауны в Бресте" },
        { id: 4, text: "Бани и Сауны в Витебске" },
        { id: 5, text: "Бани и Сауны в Гродно" },
        { id: 6, text: "Бани и Сауны в Могилеве" },
      ],
    },
    {
      id: 10,
      text: "Авто напрокат",
      list: [
        { id: 1, text: "Авто напрокат в Минске" },
        { id: 2, text: "Авто напрокат в Гомеле" },
        { id: 3, text: "Авто напрокат в Бресте" },
        { id: 4, text: "Авто напрокат в Витебске" },
        { id: 5, text: "Авто напрокат в Гродно" },
        { id: 6, text: "Авто напрокат в Могилеве" },
      ],
    },
  ];

  useEffect(() => {
    let cookieUser = getCookie("user");

    if (!cookieUser) return;
    setCookie(cookieUser);
  }, []);

  return (
    <>
      <nav className={styles.navbar}>
        <ul className={styles.list}>
          {link.map((elem, index) => (
            <li
              className={styles.item}
              onClick={() => setActiveId(elem.id)}
              key={index}>
              <Link
                className={clsx(
                  styles.text,
                  activeId === elem.id && styles["text--active"]
                )}
                href={elem.href || "./"}>
                {elem.sprite && (
                  <span className={styles["sprite-margin"]}>
                    <Sprite
                      id={elem.sprite}
                      colour={activeId === elem.id ? "black" : "#8291a3"}
                    />
                  </span>
                )}
                {elem.text}
              </Link>
              <div
                className={clsx(
                  styles.focus,
                  activeId === elem.id && styles["focus--active"]
                )}></div>
            </li>
          ))}
        </ul>

        {cookie ? (
          <p className={styles.login}>{cookie}</p>
        ) : (
          <ul className={styles["list-size"]}>
            <li
              className={styles.item}
              onClick={() => setActiveId(bookmark.id)}>
              <Link
                href={"./"}
                className={clsx(
                  styles.text,
                  activeId === bookmark.id && styles["text--active"]
                )}>
                {bookmark.text}
                <span className={styles["margin-heart"]}>
                  <Sprite
                    id="heart"
                    colour={
                      (activeId === bookmark.id && "#1e2123") || "#8291a3"
                    }
                  />
                </span>
              </Link>

              <div
                className={clsx(
                  styles.focus,
                  activeId === bookmark.id && styles["focus--active"]
                )}></div>
            </li>
            <li className={styles.item}>
              <Link
                href={"./authorization"}
                className={clsx(styles.login, styles.login)}>
                Вход и регистрация
              </Link>
            </li>
          </ul>
        )}
      </nav>

      <nav className={styles.navigation}>
        <Image src="/logo.png" height="20" width="134" alt="logo" />

        <ul className={styles["navigation-list"]}>
          <Select
            massive={underList[0]}
            active={activeId}
            setActive={setActiveId}
            underlining={true}
          />
          <Select
            massive={underList[1]}
            active={activeId}
            setActive={setActiveId}
            underlining={true}
          />
          <Select
            massive={underList[2]}
            active={activeId}
            setActive={setActiveId}
            underlining={true}
          />
          <Select
            massive={underList[3]}
            active={activeId}
            setActive={setActiveId}
            underlining={true}
          />
        </ul>

        <button className={styles.button}>+ Разместить объявление</button>
      </nav>
    </>
  );
};
