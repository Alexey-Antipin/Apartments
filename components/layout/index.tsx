import styles from "./Layout.module.scss";
import { useRouter } from "next/router";
import { LayoutProps } from "../../ts";
import { Footer } from "../footer";
import { Header } from "../header";
import Head from "next/head";

export const Layout = ({ children }: LayoutProps) => {
  const router = useRouter();
  const url = router.asPath;
  
  return (
    <div className={styles.wrapper}>
      <Head>
        <link rel="shortcut icon" href="/static/favicon.ico" />
      </Head>

      {url !== "/registration" && url !== "/authorization" ? (
        <>
          <Header />
          {children}
          <Footer />
        </>
      ) : (
        <>{children}</>
      )}
    </div>
  );
};

export default Layout;
