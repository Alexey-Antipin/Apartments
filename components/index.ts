import { Context } from "./context";
import { Layout } from "./layout";
import { Header } from "./header";
import { Footer } from "./footer";
import { Error } from "./error";
import { Home } from "./home";

export { Context, Layout, Header, Footer, Error, Home };
