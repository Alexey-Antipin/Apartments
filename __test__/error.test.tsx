import { render } from "@testing-library/react";
import { Error } from "../components/error";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "./",
      pathname: "",
      query: "",
      asPath: "/",
      push: jest.fn(),
    };
  },
}));

describe("Error", () => {
  test("heading", () => {
    const { getByRole } = render(<Error />);
    let heading = getByRole("heading", { name: /Ошибка 404/i });
    expect(heading).toBeInTheDocument();
  });
  test("text", () => {
    const { getByText } = render(<Error />);
    let text = getByText(/Вернуться на главную/i);
    expect(text).toBeInTheDocument();
  });
  test("component snapshot", () => {
    const component = render(<Error />);
    expect(component).toMatchSnapshot();
  });
});
