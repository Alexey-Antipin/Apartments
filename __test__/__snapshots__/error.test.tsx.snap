// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="error"
      >
        <div
          class="wrapper"
        >
          <img
            alt="points"
            class="picture"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="/_next/image?url=%2Fpoints.png&w=128&q=75"
            srcset="/_next/image?url=%2Fpoints.png&w=64&q=75 1x, /_next/image?url=%2Fpoints.png&w=128&q=75 2x"
            style="color: transparent;"
            width="60"
          />
          <div>
            <h2
              class="title"
            >
              Ошибка 404
            </h2>
            <p
              class="text"
            >
              Возможно, у вас опечатка в адресе страницы, или её просто не существует
            </p>
            <a
              class="button"
              href="/"
            >
              <svg
                fill="none"
                height="12"
                viewBox="0 0 12 12"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.7984 5.36427L6.41443 0.458394C6.17811 0.243027 5.82174 0.243051 5.58551 0.458371L0.201486 5.3643C0.0121814 5.5368 -0.0503497 5.80258 0.0421345 6.04138C0.134642 6.28019 0.359876 6.43448 0.615977 6.43448H1.4759V11.3498C1.4759 11.5447 1.63391 11.7027 1.8288 11.7027H4.7799C4.97478 11.7027 5.1328 11.5447 5.1328 11.3498V8.36537H6.86721V11.3498C6.86721 11.5447 7.02523 11.7027 7.22011 11.7027H10.1711C10.366 11.7027 10.524 11.5447 10.524 11.3498V6.43448H11.3841C11.6401 6.43448 11.8654 6.28016 11.9579 6.04138C12.0503 5.80256 11.9877 5.5368 11.7984 5.36427Z"
                  fill="#1E2123"
                />
              </svg>
              <span
                class="button-text"
              >
                Вернуться на главную
              </span>
            </a>
          </div>
          <div>
            <svg
              fill="none"
              height="273"
              viewBox="0 0 613 273"
              width="613"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                filter="url(#filter0_d_552_2303)"
                opacity="0.3"
              >
                <path
                  d="M30 201.328H128.868V237.481H170.507V201.328H196.053V166.481H170.507V31.8119H115.994L30 167.083V201.328ZM129.672 166.481H73.8518V164.874L128.063 79.2122H129.672V166.481Z"
                  fill="white"
                />
                <path
                  d="M305.834 242C357.631 242 388.911 202.634 389.011 134.847C389.112 67.5629 357.43 29 305.834 29C254.137 29 222.757 67.4625 222.656 134.847C222.455 202.433 253.936 241.9 305.834 242ZM305.834 205.948C282.198 205.948 266.91 182.248 267.011 134.847C267.111 88.1499 282.299 64.6507 305.834 64.6507C329.268 64.6507 344.556 88.1499 344.556 134.847C344.657 182.248 329.369 205.948 305.834 205.948Z"
                  fill="white"
                />
                <path
                  d="M416.947 201.328H515.814V237.481H557.453V201.328H583V166.481H557.453V31.8119H502.94L416.947 167.083V201.328ZM516.619 166.481H460.798V164.874L515.01 79.2122H516.619V166.481Z"
                  fill="white"
                />
              </g>
              <defs>
                <filter
                  color-interpolation-filters="sRGB"
                  filterUnits="userSpaceOnUse"
                  height="273"
                  id="filter0_d_552_2303"
                  width="613"
                  x="0"
                  y="0"
                >
                  <feflood
                    flood-opacity="0"
                    result="BackgroundImageFix"
                  />
                  <fecolormatrix
                    in="SourceAlpha"
                    result="hardAlpha"
                    type="matrix"
                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                  />
                  <feoffset
                    dy="1"
                  />
                  <fegaussianblur
                    stdDeviation="15"
                  />
                  <fecolormatrix
                    type="matrix"
                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                  />
                  <feblend
                    in2="BackgroundImageFix"
                    mode="normal"
                    result="effect1_dropShadow_552_2303"
                  />
                  <feblend
                    in="SourceGraphic"
                    in2="effect1_dropShadow_552_2303"
                    mode="normal"
                    result="shape"
                  />
                </filter>
              </defs>
            </svg>
          </div>
          <img
            alt="points"
            class="picture-white"
            data-nimg="1"
            decoding="async"
            height="60"
            loading="lazy"
            src="/_next/image?url=%2Fwhite-points.png&w=128&q=75"
            srcset="/_next/image?url=%2Fwhite-points.png&w=64&q=75 1x, /_next/image?url=%2Fwhite-points.png&w=128&q=75 2x"
            style="color: transparent;"
            width="60"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="error"
    >
      <div
        class="wrapper"
      >
        <img
          alt="points"
          class="picture"
          data-nimg="1"
          decoding="async"
          height="60"
          loading="lazy"
          src="/_next/image?url=%2Fpoints.png&w=128&q=75"
          srcset="/_next/image?url=%2Fpoints.png&w=64&q=75 1x, /_next/image?url=%2Fpoints.png&w=128&q=75 2x"
          style="color: transparent;"
          width="60"
        />
        <div>
          <h2
            class="title"
          >
            Ошибка 404
          </h2>
          <p
            class="text"
          >
            Возможно, у вас опечатка в адресе страницы, или её просто не существует
          </p>
          <a
            class="button"
            href="/"
          >
            <svg
              fill="none"
              height="12"
              viewBox="0 0 12 12"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.7984 5.36427L6.41443 0.458394C6.17811 0.243027 5.82174 0.243051 5.58551 0.458371L0.201486 5.3643C0.0121814 5.5368 -0.0503497 5.80258 0.0421345 6.04138C0.134642 6.28019 0.359876 6.43448 0.615977 6.43448H1.4759V11.3498C1.4759 11.5447 1.63391 11.7027 1.8288 11.7027H4.7799C4.97478 11.7027 5.1328 11.5447 5.1328 11.3498V8.36537H6.86721V11.3498C6.86721 11.5447 7.02523 11.7027 7.22011 11.7027H10.1711C10.366 11.7027 10.524 11.5447 10.524 11.3498V6.43448H11.3841C11.6401 6.43448 11.8654 6.28016 11.9579 6.04138C12.0503 5.80256 11.9877 5.5368 11.7984 5.36427Z"
                fill="#1E2123"
              />
            </svg>
            <span
              class="button-text"
            >
              Вернуться на главную
            </span>
          </a>
        </div>
        <div>
          <svg
            fill="none"
            height="273"
            viewBox="0 0 613 273"
            width="613"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              filter="url(#filter0_d_552_2303)"
              opacity="0.3"
            >
              <path
                d="M30 201.328H128.868V237.481H170.507V201.328H196.053V166.481H170.507V31.8119H115.994L30 167.083V201.328ZM129.672 166.481H73.8518V164.874L128.063 79.2122H129.672V166.481Z"
                fill="white"
              />
              <path
                d="M305.834 242C357.631 242 388.911 202.634 389.011 134.847C389.112 67.5629 357.43 29 305.834 29C254.137 29 222.757 67.4625 222.656 134.847C222.455 202.433 253.936 241.9 305.834 242ZM305.834 205.948C282.198 205.948 266.91 182.248 267.011 134.847C267.111 88.1499 282.299 64.6507 305.834 64.6507C329.268 64.6507 344.556 88.1499 344.556 134.847C344.657 182.248 329.369 205.948 305.834 205.948Z"
                fill="white"
              />
              <path
                d="M416.947 201.328H515.814V237.481H557.453V201.328H583V166.481H557.453V31.8119H502.94L416.947 167.083V201.328ZM516.619 166.481H460.798V164.874L515.01 79.2122H516.619V166.481Z"
                fill="white"
              />
            </g>
            <defs>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="273"
                id="filter0_d_552_2303"
                width="613"
                x="0"
                y="0"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="1"
                />
                <fegaussianblur
                  stdDeviation="15"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="effect1_dropShadow_552_2303"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_552_2303"
                  mode="normal"
                  result="shape"
                />
              </filter>
            </defs>
          </svg>
        </div>
        <img
          alt="points"
          class="picture-white"
          data-nimg="1"
          decoding="async"
          height="60"
          loading="lazy"
          src="/_next/image?url=%2Fwhite-points.png&w=128&q=75"
          srcset="/_next/image?url=%2Fwhite-points.png&w=64&q=75 1x, /_next/image?url=%2Fwhite-points.png&w=128&q=75 2x"
          style="color: transparent;"
          width="60"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
